nightingale_dispatcher/src/nightingale_dispatcher/navigate_task.py

```diff
@@ -1,18 +1,30 @@
 #!/usr/bin/env python3
 
 import rospy
-
 from nightingale_dispatcher.task import Task
+import actionlib
+from actionlib_msgs.msg import GoalStatus
+from nightingale_msgs.msg import RoomRunnerAction, RoomRunnerActionGoal
 
 
 class NavigateTask(Task):
-    # Go to position with given context
-    # TODO define parameters
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError()
+    def __init__(self):
+        action_name = "/nightingale_navigation/room_runner"
+        self.action_client = actionlib.SimpleActionClient(action_name, RoomRunnerAction)
+        rospy.loginfo(f"Waiting for {action_name} action server")
+        if self.action_client.wait_for_server():
+            rospy.loginfo(
+                f"Found the {action_name} action server",
+            )
+        else:
+            rospy.logfatal(
+                f"Failed to find the {action_name} action server",
+            )
 
-    # TODO define inputs
-    def execute(self, *args, **kwargs):
-        # Check if there are intermediate steps to travel to the destination
-        # Call room_runner to drive to pose
-        raise NotImplementedError()
+    def execute(self, room_number, sublocation):
+        goal = RoomRunnerActionGoal()
```

Comment:
Can this be a RoomRunnerGoal so that lines 26 is simpler: `goal.room_number = room_number`? I recall doing something similar with the arms

Reply:
hmm should mission_planner be building the action specific types? i think its cleaner if the Task handles all the action specific stuff. the task converts its input arguments to the ActionGoal type. otherwise, mission_planner might get messy with all the action specific types